lib/ClusterShell/Worker/Rsh.py

```diff
@@ -59,8 +60,31 @@ def _build_cmd(self):
         cmd_l.append("%s" % self.key)  # key is the node
         cmd_l.append("%s" % self.command)
 
+        # rsh does not properly return exit status
+        # force the exit status to be printed out
+        cmd_l.append("; echo XXRETCODE: $?")
+
         return (cmd_l, None)
 
+    def _on_nodeset_start(self, nodes):
+        """Override _on_nodeset_start to guarantee that rsh_rc is initialized """
+        self.rsh_rc = None
+        ExecClient._on_nodeset_start(self, nodes)
```

Comment:
Usually attribute declaration should be done in `__init__()`. That's ok like that, but, if you happen to refresh this code, please move that to __init__

Reply:
Agree.  Just seemed like overkill to override init().  In this case it seemed cleaner to do this, but I will move it to __init__(), since it really is the right thing to do!

Reply:
Done!